FlowCrypt/Functionality/DataManager/Encrypted Storage/KeyChainService.swift

```diff
@@ -10,64 +10,60 @@ import FlowCryptCommon
 import Foundation
 import Security
 
-// keychain is used to generate and retrieve encryption key which is used to encrypt local DB
-// it does not contain any actual data or keys other than the db encryption key
+/// keychain is used to generate and retrieve encryption key which is used to encrypt local DB
+/// it does not contain any actual data or keys other than the db encryption key
+/// index of the keychain entry is dynamic (set up once per app installation), set in user defaults
+actor KeyChainService {
 
-protocol KeyChainServiceType {
-    func getStorageEncryptionKey() throws -> Data
-}
-
-struct KeyChainService: KeyChainServiceType {
-    private static var logger = Logger.nested(in: Self.self, with: "Keychain")
+    private let logger = Logger.nested("KeyChain")
+    private let keyByteLen = 64
 
```

Comment:
This class was rewritten without changing functionality. Just general cleanup (used to be somewhat static). 

Now an actor.